libraries/src/Plugin/CMSPlugin.php

```diff
@@ -47,58 +51,62 @@ abstract class CMSPlugin implements DispatcherAwareInterface, PluginInterface
      * @var    string
      * @since  1.5
      */
+    //phpcs:disable PSR2.Classes.PropertyDeclaration.Underscore
     protected $_name = null;
+    //phpcs:enable PSR2.Classes.PropertyDeclaration.Underscore
 
     /**
      * The plugin type
      *
      * @var    string
      * @since  1.5
      */
+    //phpcs:disable PSR2.Classes.PropertyDeclaration.Underscore
     protected $_type = null;
+    //phpcs:enable PSR2.Classes.PropertyDeclaration.Underscore
 
     /**
      * Affects constructor behavior. If true, language files will be loaded automatically.
      *
      * @var    boolean
      * @since  3.1
+     * @deprecated 5.0 Use the LanguageAwareTrait and call loadLanguage() in doInitialise().
      */
     protected $autoloadLanguage = false;
 
     /**
-     * Should I try to detect and register legacy event listeners, i.e. methods which accept unwrapped arguments? While
-     * this maintains a great degree of backwards compatibility to Joomla! 3.x-style plugins it is much slower. You are
-     * advised to implement your plugins using proper Listeners, methods accepting an AbstractEvent as their sole
-     * parameter, for best performance. Also bear in mind that Joomla! 5.x onwards will only allow proper listeners,
-     * removing support for legacy Listeners.
+     * Should we allow "magic" late initialisation of this plugin using the doInitialise code?
      *
-     * @var    boolean
-     * @since  4.0.0
-     *
-     * @deprecated
+     * @var   bool
+     * @since  __DEPLOY_VERSION__
      */
-    protected $allowLegacyListeners = true;
+    protected $allowLateInitialisation = true;
 
     /**
-     * The application object
-     *
-     * @var    CMSApplicationInterface
+     * Flag for the initialisePlugin method.
      *
-     * @since  4.2.0
+     * @var    bool
+     * @since  __DEPLOY_VERSION__
      */
-    private $application;
+    private $isPluginInitialised = false;
 
     /**
-     * Constructor
+     * Constructor.
+     *
+     * Do not put any slow initialisation code in it, e.g. code which accesses the database,
+     * performs lengthy calculations, or calls external services over HTTP. Put this code into the
+     * doInitialise() method which is called ONCE, before the first event handler in your plugin is
+     * executed.
```

Comment:
```suggestion
     * Do not put any slow initialisation code in the constructor, e.g. code which accesses the database,
     * performs lengthy calculations, or calls external services over HTTP. Put that code into the
     * doInitialise() method which is called ONCE, before the first event handler in your plugin is
     * executed.
```

clarifies what "it" and "this code"